client/components/OperationList.js

```diff
@@ -56,24 +56,27 @@ class LabelComponent extends React.Component {
 
     onBlur() {
         let customLabel = this.dom().value;
-        if (customLabel && customLabel !== this.defaultValue()) {
-            if (customLabel.trim().length) {
+        if (customLabel) {
+            // If the new non empty customLabel value is different from the current one, save it.
+            if (customLabel !== this.defaultValue() && customLabel.trim().length) {
```

Comment:
Sorry for getting back to this, but I realized we could do slightly better, so as not to save customLabels which are the default value plus a few spaces:

`if (customLabel.trim().length && customLabel.trim() !== this.defaultValue())`